pkg/storage/storage.go

```diff
@@ -143,17 +161,26 @@ func (s *localStorage[T]) unsafeLength() uint {
 	return uint(length)
 }
 
-func (s *localStorage[T]) sendMonitoringData(id string, o T, isDeletion bool, count uint) {
+func (s *localStorage[T]) sendMonitoringData(id string, o T, eventType EventType, count uint) {
 	if s.measurement != "" {
 		p := influxdb2.NewPointWithMeasurement(s.measurement)
 		p.AddTag("id", id)
-		p.AddTag("isDeletion", strconv.FormatBool(isDeletion))
+		p.AddTag("event_type", string(eventType))
 		p.AddField("count", count)
 
 		if s.callback != nil {
-			s.callback(p, o, isDeletion)
+			s.callback(p, o, eventType)
 		}
 
 		monitoring.WriteInfluxPoint(p)
 	}
 }
+
+func (s *localStorage[T]) periodicallySendMonitoringData(d time.Duration) {
+	ctx := context.Background()
+	for ctx.Err() == nil {
+		stub := new(T)
+		s.sendMonitoringData("", *stub, Periodically, s.Length())
+		time.Sleep(d)
```

Comment:
Is `sleep` appropriate in Go? In other systems, the usage of sleep is usually discouraged. In many cases, developers need to be careful not to send the current thread to sleep (which would also block other executions). Therefore, many have some reactivation, timer, ... concept. According to my search, Go would also feature a [Ticker](https://pkg.go.dev/time#Ticker) struct respectively [NewTicker](https://pkg.go.dev/time#NewTicker) function.

Reply:
In Go we just sleep the current Go routine and this way the scheduler does not block other executions.
I searched for other occurrences of `sleep` in our code and found only few. Instead we often use `<-time.After(...)`. I expect those variants to be similar but if you want we can use this instead?
With the Ticker we would also wait and receive from a channel `<-ticker.C` but with the overhead of Creating the ticker before and stopping it afterwards.

Reply:
Thanks for digging into this, I was also reading some more. A nice [comparison is on StackOverflow](https://stackoverflow.com/a/64500114), but there is no "clear" result. One improvement described there is the ability to cancel `time.After` immediately, which I think is nice (and more responsive?). If you don't have a preference, I would suggest to use `time.After` (as we have done in the code already, `time.sleep` has so far only been used in tests).